include/inviwo/core/processors/processor.h

```diff
@@ -397,6 +395,39 @@ class IVW_CORE_API Processor : public PropertyOwner,
 
 inline ProcessorNetwork* Processor::getNetwork() const { return network_; }
 
+// Assume ownership of port, needed for dynamic ports
+template <typename T, typename std::enable_if_t<std::is_base_of<Inport, T>::value, int>>
+T& Processor::addPort(std::unique_ptr<T> port, const std::string& portGroup) {
+    T& ret = *port;
+    addPortInternal(port.get(), portGroup);
+    ownedInports_.push_back(std::move(port));
+    return ret;
+}
+
+// Assume ownership of port, needed for dynamic ports
+template <typename T, typename std::enable_if_t<std::is_base_of<Outport, T>::value, int>>
+T& Processor::addPort(std::unique_ptr<T> port, const std::string& portGroup) {
+    T& ret = *port;
+    addPortInternal(port.get(), portGroup);
+    ownedOutports_.push_back(std::move(port));
+    return ret;
+}
+
+/**
+ * Add port to processor.
+ * @note Port group is a concept for event propagation. Currently only used for
+ * ResizeEvents, which only propagate from outports to inports in the same port group
+ * @param port to add
+ * @param portGroup name of group to propagate events through (defaults to "default")
+ */
+template <typename T>
```

Comment:
Documentation should be moved to definition

Reply:
It was duplicated documentation, forgot to remove it after moving the impl out of the class declaration